examples/07_advanced/plot_mask_large_fmri.py

```diff
@@ -0,0 +1,412 @@
+by a number of binary masks, all in parallel.
+
+The issue when trying to process a large fMRI image in parallel like this is
+that each parallel process will load the entire fMRI image into memory. This
+can lead to a significant increase in memory usage and can slow down the
+processing.
+
+We will compare three different methods to mask the data from the fMRI image:
+
+1. Using the :class:`~nilearn.maskers.NiftiMasker`
+2. Using numpy indexing
+3. Using numpy indexing with shared memory
+
+"""
+
+# %%
+# Create a large fMRI image
+# -------------------------
+# Here we will create a "large" fMRI image by fetch 5 subjects'
+# fMRI images via the :func:`~nilearn.datasets.fetch_development_fmri`
+# function, concatenating them and then saving to a file.
+
+from pathlib import Path
+
+from nilearn.datasets import fetch_development_fmri
+from nilearn.image import concat_imgs
+
+N_SUBJECTS = 5
+N_REGIONS = 6
+
+fmri_data = fetch_development_fmri(n_subjects=N_SUBJECTS)
+fmri_img = concat_imgs(fmri_data.func)
+
+output_dir = Path.cwd() / "results" / "plot_large_fmri_img"
+output_dir.mkdir(parents=True, exist_ok=True)
+print(f"Large fmri file will be saved to:\n{output_dir}")
+
+fmri_path = output_dir / "large_fmri.nii.gz"
+fmri_img.to_filename(fmri_path)
+
+# %%
+# Create a set of binary masks
+# ----------------------------
+# We will now create 6 binary masks from a brain atlas. Here we will use the
+# DIFUMO atlas via the :func:`~nilearn.datasets.fetch_atlas_difumo` function.
+# This atlas is a probabilistic atlas so we will threshold it to create binary
+# masks. You might want to use a different atlas or create your own masks.
+
+from nilearn.datasets import fetch_atlas_difumo
+from nilearn.image import (
+    index_img,
+    iter_img,
+    load_img,
+    new_img_like,
+    resample_to_img,
+)
+
+atlas_path = fetch_atlas_difumo(dimension=64).maps
+
+masks = load_img(atlas_path)
+# only keep the first 6 regions
+masks = index_img(masks, slice(0, N_REGIONS))
+
+# %%
+# In this case, the atlas and the mask have different resolutions. So we will
+# resample the mask to the fMRI image. It is important to do that because
+# only :class:`~nilearn.maskers.NiftiMasker` can handle the resampling of the
+# mask to the fMRI image but not the numpy indexing method we will use later.
+
+mask_paths = []
+mask_imgs = []
+for i, mask in enumerate(iter_img(masks)):
+    resampled_mask = resample_to_img(
+        mask,
+        fmri_path,
+        interpolation="nearest",
+        copy_header=True,
+        force_resample=True,
+    )
+    data = resampled_mask.get_fdata()
+    data[data != 0] = 1
+    resampled_mask = new_img_like(
+        ref_niimg=resampled_mask,
+        data=data,
+        affine=resampled_mask.affine,
+        copy_header=True,
+    )
+    mask_imgs.append(resampled_mask)
+
+    path = output_dir / f"mask_{i}.nii.gz"
+    resampled_mask.to_filename(path)
+    mask_paths.append(path)
```

Comment:
```suggestion
masks = datasets.fetch_atlas_basc_multiscale_2015(resolution=7).maps
mask_img = nib.load(masks)
resampled_mask = resample_to_img(
    mask_img,
    fmri_path,
    interpolation="nearest",
    copy_header=True,
    force_resample=True,
)

for idx in range(N_REGIONS):
    mask = (mask_img.get_fdata() == idx)
    resampled_mask = image.new_img_like(
        ref_niimg=fmri_path,
        data=mask,
        affine=mask_img.affine,
        copy_header=True,
    )
    
    mask_imgs.append(resampled_mask)
    path = output_dir / f"mask_{i}.nii.gz"
    resampled_mask.to_filename(path)
    mask_paths.append(path)
```

Reply:
Note that this changes the atlas we use, as I think a deterministic atlas is a little closer to the common use case (and so easier to understand !)